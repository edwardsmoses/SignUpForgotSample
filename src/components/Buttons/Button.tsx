import React from 'react';
import { StyleSheet, TouchableOpacity, Text, View } from 'react-native';

import * as Colors from '../../styles/Color';

interface ButtonProps {
    onPress: () => {};
    title: string;
    height?: number;
    width?: number;
}

const Button: React.FC<ButtonProps> = ({
    title,
    onPress,
    height = 50,
    width = 300,
}) => {
    return (
        <View style={styles.containerStyle}>
            <TouchableOpacity
                onPress={onPress}
                style={[{ height: height, width: width }, styles.buttonStyle]}>
                <Text style={styles.textStyle}>{title}</Text>
            </TouchableOpacity>
        </View>
    );
};

export default Button;

const styles = StyleSheet.create({
    containerStyle: {
        alignContent: 'center',
        justifyContent: 'center',
    },
    buttonStyle: {
        backgroundColor: Colors.Blue,
        borderRadius: 15,
        alignContent: 'center',
        justifyContent: 'center',
    },
    textStyle: {
        color: 'white',
        fontSize: 20,
        textAlign: 'center',
    },
});
