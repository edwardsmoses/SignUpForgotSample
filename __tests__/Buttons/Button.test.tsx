import { render, cleanup } from 'react-native-testing-library';
import React from 'react';
import Button from '../../src/components/Buttons/Button';

afterEach(cleanup);

describe('<Button />', () => {
    it('should match snapshot', () => {
        const rendered = render(
            <Button
                onPress={() => {
                    return {};
                }}
                title="Sign Up"
            />,
        ).toJSON();

        expect(rendered).toMatchSnapshot();
    });

    // it('should properly render the text', () => {
    //     const rendered = render(
    //         <Button
    //             onPress={() => {
    //                 return {};
    //             }}
    //             title={'Sign Up'}
    //         />,
    //     );
    //     const textComponent = rendered.getByTestId('text');

    //     expect(textComponent.props.children).toEqual('Sign Up');
    // });
});
