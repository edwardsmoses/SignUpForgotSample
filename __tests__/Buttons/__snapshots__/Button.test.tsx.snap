// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "backgroundColor": "#53a1e8",
        "borderRadius": 15,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      Sign Up
    </Text>
  </View>
</View>
`;
